include/JSFunctionProxy.hh

```diff
@@ -0,0 +1,66 @@
+/**
+ * @file JSFunctionProxy.hh
+ * @author Caleb Aikens (caleb@distributive.network)
+ * @brief JSFunctionProxy is a custom C-implemented python type. It acts as a proxy for JSFunctions from Spidermonkey, and behaves like a function would.
+ * @version 0.1
+ * @date 2023-09-28
+ *
+ * Copyright (c) 2023 Distributive Corp.
+ *
+ */
+
+#ifndef PythonMonkey_JSFunctionProxy_
+#define PythonMonkey_JSFunctionProxy_
+
+#include <jsapi.h>
+
+#include <Python.h>
+/**
+ * @brief The typedef for the backing store that will be used by JSFunctionProxy objects. All it contains is a pointer to the JSFunction
+ *
+ */
+typedef struct {
+  PyObject_HEAD
+  JS::PersistentRootedObject *jsFunc;
```

Comment:
maybe this should not be a pointer?

Reply:
I think its fine, PersistentRootedObject is fine to store on the heap